source/server-discovery-and-monitoring/server-discovery-and-monitoring.rst

```diff
@@ -1349,6 +1465,26 @@ clear the server's connection pool if and only if the error is
 error messages to detect "not master" and "node is recovering"`_, and `other
 transient errors`_ and `Why close connections when a node is shutting down?`_.)
 
+Reducing race conditions in error handling
```

Comment:
I think this section needs to be defined above the pseudocode.  Maybe it should just be rolled into the https://github.com/mongodb/specifications/blob/master/source/server-discovery-and-monitoring/server-discovery-and-monitoring.rst#not-master-and-node-is-recovering section, in the prose before the pseudocode?

Reply:
Also, it doesn't seem like we have any way to test this.  Is that correct?

Reply:
I moved the generation number rule (which is required on both "network errors" and State Change Errors) to a new section above called "Application Errors".  I moved the topologyVersion rule to the not-master-and-node-is-recovering section.